hearts/hearts.py

```diff
@@ -215,40 +215,42 @@ def play_card(self, player, card):
         else:
             raise ValueError("Invalid play: it's not your turn")
 
-    def trick_points(self, trick):
-        if len(trick) < 4:
-            return 0
-        else:
-            points = 0
-            for data in trick.cards_played:
-                if data[1].suit == 'h':
-                    points += 1
-                elif data[1] == Card('Q', 's'):
-                    points += 13
-            return points
-
     def current_scores(self):
         '''
         Returns {player: int}.  Calculates the current number of points
         for all tricks completed so far. Assumes the basic Hearts rules.
         '''
         scores = {player: 0 for player in self.players}
         for trick in self.tricks:
-            scores[trick.winner()] += self.trick_points(trick)
+            scores[trick.winner()] += trick.points()
         return scores
 
     def shot_the_moon(self):
         '''
         Returns {player: Bool}.  Bool is True if and only if player's
-        score is 26 (based on the standard Hearts rules.)
+        score is 26 (based on the standard Hearts rules).  Function can
+        be called at any time during the round.
         '''
         d = {player: False for player in self.players}
-        pass
+        scores = self.current_scores()
+        for player in self.players:
+            if scores[player] == 26:
+                d[player] = True
+                return d
+        return d
 
-    def final_scores(self):
+    def final_scores(self):     # FIXME:  test_final_scores_shot_the_moon() fails. 
         scores = self.current_scores()
-        if len(self.tricks) == 13 & len(self.tricks[-1] == 4):
-            pass
+        shoot_successes = self.shot_the_moon()
+        if all(shoot_successes.values()) is False:
+            return scores
+        else:
+            for (player, attempt) in shoot_successes.items():
+                if attempt is True:
+                    scores[player] = scores[player] - 26
```

Comment:
I see. This was a little confusing for me. Would it make sense to just set the scores with assignment instead of arithmetic? 

```
if attempt:
   scores[player] = 0
else:
   sores[player] = 26
```